ssl/t1_lib.c

```diff
@@ -1164,29 +1319,45 @@ int ssl_setup_sig_algs(SSL_CTX *ctx)
          * could be that the signature is available, and the hash is available
          * independently - but not as a combination. We ignore this for now.
          */
-        if (lu->hash != NID_undef
-                && ctx->ssl_digest_methods[lu->hash_idx] == NULL) {
+        if (cache[i].hash != NID_undef
+                && ctx->ssl_digest_methods[cache[i].hash_idx] == NULL) {
```

Comment:
We really need to find a better way to do this. @levitte has proposed that the signature algorithms in the provider are actually the sig/hash pair which would simplify all of this significantly.

Reply:
Yes.  But for now, I have to work with the way it is now.